lib/Parse/Lexer.cpp

```diff
@@ -1843,9 +1843,30 @@ void Lexer::lexStringLiteral(unsigned CustomDelimiterLen) {
   assert((QuoteChar == '"' || QuoteChar == '\'') && "Unexpected start");
 
   bool IsMultilineString = advanceIfMultilineDelimiter(CurPtr, Diags);
-  if (IsMultilineString && *CurPtr != '\n' && *CurPtr != '\r')
-    diagnose(CurPtr, diag::lex_illegal_multiline_string_start)
-        .fixItInsert(Lexer::getSourceLoc(CurPtr), "\n");
+  if (IsMultilineString && *CurPtr != '\n' && *CurPtr != '\r') {
+    // Test for single-line string literals that may resemble multiline delimiter.
+    for (const char *Ptr = CurPtr; Ptr <= BufferEnd-CustomDelimiterLen; Ptr++) {
+      if (*Ptr == '\r' || *Ptr == '\n') {
+        break;
+      }
+      if (*Ptr == '#') {
+        const char *TmpPtr = Ptr + 1;
+        while (*TmpPtr == '#') {
+          TmpPtr++;
+        }
+        if (TmpPtr-Ptr == CustomDelimiterLen) {
```

Comment:
You should still be able to use `delimiterMatches()`. Try `delimiterMatches(CustomDelimiterLen, TmpPtr, nullptr)`